### ISSUE TITLE
Read this before submitting 0.4.6.0 bootloop issue.

### ISSUE THREAD

**Issue by JohnGLFour** [ElderDrivers/EdXposed#384]
With 0.4.6.0 they moved the SEpolicy from the post-fs-data.sh to the custom magisk build. Even readme.md said that you need to install custom Magisk build before you install the 0.4.6.0 Edxposed.

TL;DR: Try install custom Magisk before submitting the issue.

**Comment by JohnGLFour** [comment]
Thanks for adding the "good first issue" tag. The reason I posted this is because people saying that 0.4.6.0 causes bootloop, but in reality they didn't install the custom magisk build. (Can confirm this as I tried installing 0.4.6.0 in regular magisk, and it bootloops)

**Comment by p1r473** [comment]
Hi, I got myself in a pickle
Im on a Pixel 4 XL. I accidently installed EdXposed 0.4.5.1_beta instead of 0.4.6 and it caused a bootloop
I was able to get back into the OS by reflashing the latest OTA update, but I am left with a problem:
Any time I reflash Magisk, it recreates /data/adb/modules/riru-core and riru_edxposed_sandhook , the old versions that cause bootloop
No matter how many times I go in and delete these folders with ADB, the ygety recreated any time I reflash Magisk

How can I prevent Magisk from reinstalling these modules every boot? Please help!

**Comment by ecompton59** [comment]
flash magisk manager for recovery then go to terminal in twrp type mm load
core mode it will boot then delete the bad magisk mod turn off core mode
you'll be good then

On Saturday, November 9, 2019, p1r473 <notifications@github.com> wrote:
> Hi, I got myself in a pickle
> Im on a Pixel 4 XL. I accidently installed EdXposed 0.4.5.1_beta instead
of 0.4.6 and it caused a bootloop
> I was able to get back into the OS by reflashing the latest OTA update,
but I am left with a problem:
> Any time I reflash Magisk, it recreates /data/adb/modules/riru-core and
riru_edxposed_sandhook , the old versions that cause bootloop
> No matter how many times I go in and delete these folders with ADB, the
ygety recreated any time I reflash Magisk
>
> How can I prevent Magisk from reinstalling these modules every boot?
Please help!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.<
https://ci6.googleusercontent.com/proxy/roMMCzmOxpEJE5Y8QW4pPY-mAG_WH-8bOO9ctQzjHR4-71UUdWXSGz0z02m4q-XGDu6IwUVWzpRZI_GsPUGDpAvKKLVtQMm0UbtvKPjjatT0M4g887XN0bnMYklz_b1Uc8y6YRsun0QOcfri-BJbj5wmIpW9zKXHXhckmp-F_JS5BUrLC0T52xkUT6wUG4fgFZhnF7YwIa4Dab23lTa0x_2ylM4RRlZEtJlmGKkcoQ=s0-d-e1-ft#https://github.com/notifications/beacon/AHPO5JD5QGUMJLTYHZKSETLQS322DA5CNFSM4JIJ7HL2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEDULSHQ.gif>

**Comment by ecompton59** [comment]
download magisk recovery manager flash in twrp goto terminal in twrp type mm load core mode it will boot then and delete bad mods from magisk turn core mode off and you'll be good

**Comment by p1r473** [comment]
Thanks, but Pixel 4 has no TWRP
I ended up using the Core-only magisk image from XDA forums which removed all modules.

**Comment by ecompton59** [comment]
great keep that file just incase when testing lol ive been in your shoes with OnePlus 7pro on 10

**Comment by CiriousJoker** [comment]
@p1r473 
Next time, you can boot into the loading screen, connect via adb and then do:

- adb shell
- su
- touch /cache/.disable_magisk

**Comment by aliantonio** [comment]
I am unable to install the custom magisk build. When I change the update channel to "Custom" in settings and enter the URL (http://edxp.meowcat.org/repo/version.json) it hangs and fails to load the custom build.
![Screenshot_20191113-111513](https://user-images.githubusercontent.com/24923847/68782518-930f1b80-0607-11ea-94c6-7ba341dda155.png)

**Comment by NicolasChamllet** [comment]
> With 0.4.6.0 they moved the SEpolicy from the post-fs-data.sh to the custom magisk build. Even readme.md said that you need to install custom Magisk build before you install the 0.4.6.0 Edxposed.
> 
> TL;DR: Try install custom Magisk before submitting the issue.


Can u be more specific with this "custom Magisk"? what's it and where can i find it?

**Comment by JohnGLFour** [comment]
> > With 0.4.6.0 they moved the SEpolicy from the post-fs-data.sh to the custom magisk build. Even readme.md said that you need to install custom Magisk build before you install the 0.4.6.0 Edxposed.
> > TL;DR: Try install custom Magisk before submitting the issue.
> 
> Can u be more specific with this "custom Magisk"? what's it and where can i find it?

Find it in version 0.4.6.0 release page. Its mentioned

**Comment by ecompton59** [comment]
you could also use selinux switch on xda it worked on moto z2 force it makes selinux permissive after boot https://forum.xda-developers.com/android/apps-games/app-selinux-switch-t3656502 instead of different magisk that worksif selinux is permissive that fixes issue

**Comment by ecompton59** [comment]
![Screenshot_20191121-082248](https://user-images.githubusercontent.com/31387300/69341743-21068a00-0c38-11ea-9adb-65c13adc7d14.jpg)
if it dont boot use this goto twrp terminal and type mm load magisk core mode then boot

**Comment by JohnGLFour** [comment]
> you could also use selinux switch on xda it worked on moto z2 force it makes selinux permissive after boot https://forum.xda-developers.com/android/apps-games/app-selinux-switch-t3656502 instead of different magisk that worksif selinux is permissive that fixes issue

But doing so will let your device on risk

**Comment by MlgmXyysd** [comment]
pinned, now closed

**Comment by M1ST3RH** [comment]
Magisk has added Support pre-init sepolicy patch in modules in 20.3 release. So can i flash exposed 0.4.6 with official magisk now without any bootloop? 
I am using redmi note 7 pro (violet) MIUI 11.0.5 , Android Pie.

**Comment by MlgmXyysd** [comment]
@M1ST3RH 0.4.6.1

**Comment by M1ST3RH** [comment]
> @M1ST3RH 0.4.6.1

Where can i find it? Will it work without bootloop on official magisk 20.3 ?

**Comment by MlgmXyysd** [comment]
@M1ST3RH EdXposed Canary

**Comment by M1ST3RH** [comment]
> @M1ST3RH EdXposed Canary

So can i use it without any bootloop issues?

**Comment by MlgmXyysd** [comment]
@M1ST3RH yes, with Magisk 20.2+

**Comment by M1ST3RH** [comment]
> @M1ST3RH yes, with Magisk 20.2+

Thanks :)

**Comment by solquest3** [comment]
Hello.
I read here that Magisk 20.3 shouldn't bootloop after installing 0.4.6.0 Edxposed, so I've tried but I got bootloop.
So I've uninstalled Magisk and the flashed Canary Magisk 20.4.20414-e02e46d0 but again I got bootloop after installing 0.4.6.0 Edxposed.
What should I do for installing Edxposed without bootloop?
I'm on Xiaomi Redmi 4X android 9.
Thanks!
